### ISSUE TITLE
Fix FP S3800 (`function-return-type`): Make the exception of returning "this" work

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4251]
The rule reports functions with multiple return statements of different types. Also, it is supposed to make exceptions to functions whose one of its return statements returns `this`. However, the exception is not working, and a false positive is raised on the following snippet:

```javascript
String.prototype.foo = function()
{
    if (this.length == 10)
        return this.substring(5);
    
    return this;
}
```

Reported in https://community.sonarsource.com/t/javascript-s3800-fp-when-working-on-prototype-function/102118
